### PR TITLE
Support service mapping part1

### PR DESCRIPTION
**Pull request by m00g3n**
Support installing Service Mapping Operator in the Connectivity Proxy Reconciler (part 1)
- add Service Mapping Secret management

### CODE REVIEW

File: pkg/reconciler/instances/connectivityproxy/action.go [hunk 1]
```diff
@@ -45,50 +51,60 @@ func (a *CustomAction) Run(context *service.ActionContext) error {
...
-		context.Logger.Debug("Reading ServiceBinding Secret")
-		bindingSecret, err := a.Loader.FindSecret(context, binding)
+	// detect if connectivity-proxy reconciliation should not be skipped
+
+	if binding == nil && app == nil {
+		return nil
+	}
 
-		context.Logger.Debug("Service Binding Secret check")
+	// detect if connectivity-proxy is being deleted
```
Comment (koala7659) [inline pkg/reconciler/instances/connectivityproxy/action.go:60]: Please remove secret `connectivity-sm-operator-secrets-tls` when connectivity-proxy is deleted

Suggested change change `removeIstioSecrets` into `removeInstalledSecrets`

```
func (a *CommandActions) removeInstalledSecrets(context *service.ActionContext) error {

	_, err := context.KubeClient.DeleteResource(context.Context, "secret", "cc-certs", "istio-system")
	if err != nil {
		context.Logger.Error("Error during removal of cc-certs in istio-system")
		return errors.Wrap(err, "Error during removal of cc-certs in istio-system")
	}

	_, err = context.KubeClient.DeleteResource(context.Context, "secret", "cc-certs-cacert", "istio-system")
	if err != nil {
		context.Logger.Info("Error during removal of cc-certs-cacert in istio-system")
		return errors.Wrap(err, "Error during removal of cc-certs-cacert in istio-system")
	}

	_, err = context.KubeClient.DeleteResource(context.Context, "secret", "connectivity-sm-operator-secrets-tls", "kyma-system")
	if err != nil {
		context.Logger.Info("Error during removal of connectivity-sm-operator-secrets-tls in kyma-system")
		return errors.Wrap(err, "Error during removal of connectivity-sm-operator-secrets-tls in kyma-system")
	}
	return nil
}
```

File: pkg/reconciler/instances/connectivityproxy/command.go [hunk 2]
```diff
@@ -65,29 +67,53 @@ func (a *CommandActions) getChartProvider(context *service.ActionContext, withFi
...
+
+func (a *CommandActions) CreateSecretMappingOperator(s *service.ActionContext, ns string) (map[string][]byte, error) {
+	cs, err := s.KubeClient.Clientset()
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get a target cluster client set")
+	}
+
+	repo := secrets.NewSecretRepo(ns, cs)
+	data, err := ssl.GenerateCertificate(
+		"connectivity-proxy-smv.kyma-system.svc",
```
Comment (koala7659) [inline pkg/reconciler/instances/connectivityproxy/command.go:104]: connectivity-proxy-srv.kyma-system.svc

Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/command.go:104]: Turned out domain name was correct

**Review by koala7659 — DISMISSED** [review]
LGTM

**Review by akgalwas — APPROVED** [review]

**Review by janmedrek — APPROVED** [review]

**Pull request merged**